### PR TITLE
DOC: fix typos

### PR DESCRIPTION
**Pull request by luzpaz**
Found using `codespell v2.0.dev`  
```
codespell -q 3 -S ./extern,./doc/users/credits.rst,./doc/api/prev_api_changes,./doc/users/prev_whats_new -L afe,ans,axises,ba,coo,flate,fo,hist,inout,ment,nd,offsetp,oly,ot,te,thisy,vas,whis,wit
```

### CODE REVIEW

**Review by dopplershift — APPROVED** [review]
Thanks for the contribution!

**Comment by jklymak** [comment]
We periodically get PRs like this, which is great, but they are disruptive in that they hit a bunch of files trivially.  Have we given any thought to just running codespell as part of CI, just like we run flake8?

File: examples/axisartist/demo_curvelinear_grid2.py [hunk 1]
```diff
@@ -1,6 +1,6 @@
 """
 ======================
-Demo Curvelinear Grid2
+Demo Curvilinear Grid2
```
Comment (QuLogic) [inline examples/axisartist/demo_curvelinear_grid2.py:3]: This one is on purpose; see the filename and classes used. Maybe it should be `CurveLinear` in the title instead, though?

File: tutorials/toolkits/axisartist.py [hunk 2]
```diff
@@ -121,7 +121,7 @@
    :align: center
    :scale: 50
 
-   Demo Curvelinear Grid
+   Demo Curvilinear Grid
```
Comment (QuLogic) [inline tutorials/toolkits/axisartist.py:124]: Ditto

File: tutorials/toolkits/axisartist.py [hunk 3]
```diff
@@ -566,7 +566,7 @@ def inv_tr(x, y):
...
```
Comment (QuLogic) [inline tutorials/toolkits/axisartist.py:569]: Ditto.

**Review by QuLogic — COMMENTED** [review]
Can you also fix this new typo?
https://github.com/matplotlib/matplotlib/pull/18976/files#diff-33fca2ac0f7a278ffa15904a36c488500a9a4f594094e4bbffb1a10d7e9948afR30

**Comment by QuLogic** [comment]
The trouble with running it in CI is that there are a lot of false positives; I'm sure @luzpaz had to spend a while cherry picking the bits that are actually errors.

Possibly we could use reviewdog to limit the results to the diff, if codespell doesn't have a way to do that.

**Comment by luzpaz** [comment]
> The trouble with running it in CI is that there are a lot of false positives; I'm sure @luzpaz had to spend a while cherry picking the bits that are actually errors.
> Possibly we could use reviewdog to limit the results to the diff, if codespell doesn't have a way to do that.

Indeed. But each time I added them to  to the `-L` flag list. Codespell can be run specifically on the diff. I've seen it but not remembering exactly now.

File: examples/axisartist/demo_curvelinear_grid2.py (hunk 1 above)
Comment (tacaswell) [inline examples/axisartist/demo_curvelinear_grid2.py:3]:
```suggestion
Demo CurveLinear Grid2
```

File: tutorials/toolkits/axisartist.py (hunk 2 above)
Comment (tacaswell) [inline tutorials/toolkits/axisartist.py:124]:
```suggestion
   Demo CurveLinear Grid
```

File: tutorials/toolkits/axisartist.py (hunk 3 above)
Comment (tacaswell) [inline tutorials/toolkits/axisartist.py:569]:
```suggestion
   Demo CurveLinear Grid
```

**Review by tacaswell — CHANGES_REQUESTED** [review]
I agree with @QuLogic 's comment about `CureveLinear` vs `Curvilinear`.  For better or worse the class name is `CurveLinear` and we should match that in the text.

Anyone can clear this review.

**Comment by luzpaz** [comment]
Added requested, thanks @tacaswell.  
+ rebased

**Review by jklymak — APPROVED** [review]

**Review by timhoffm — APPROVED** [review]

**Comment by timhoffm** [comment]
Thanks @luzpaz!

**Pull request merged**